### DISCUSSION TITLE
Looking for a script to convert Zotero Snapshot .html files into PDF

### DISCUSSION THREAD

**Discussion by ryanjamurphy** [windingwind/zotero-actions-tags#257] [Q&A]
Hi folks, thanks for building Actions & Tags and all of these community scripts. I've just discovered them, and they look really useful!

I'd like a way to automatically (or at least effortlessly) get PDFs of items saved to Zotero when Zotero can't find any associated PDF files via the Connector. I can think of at least two ways to do this:
- Get Zotero to save Snapshots whenever this happens, then convert the Snapshot .html file to PDF
- on createItem, if there are no attachments, use the URL associated with the item to create a PDF from the webpage

Before I go down the rabbithole and try to build out one of these, though, I thought I'd ask: has anyone done it already? Or is there anything similar I can work off of?

(I have searched for ten minutes already but didn't find anything. Still, I thought I'd ask, in case I missed it.)

**Comment by windingwind** [comment]
May I ask why you want to convert the webpage to PDF? Zotero Reader actually well supports webpage natively since Zotero 7 beta.

**Reply by ryanjamurphy** [reply]
Sure! I have a whole other system for managing, reviewing, annotating, and extracting annotations from PDF across multiple devices (iOS, iPadOS, macOS). HTML files can't work with many parts of that system, unfortunately. 

I already create PDFs and attach them to Zotero publication items manually, but it takes like 7 steps after the initial invocation of Zotero Connector. So I'm ultimately looking to make those steps happen just by clicking the Zotero Connector button in the browser.

**Comment by FeralFlora** [comment]
A similar use case would be to convert attachments to Markdown, so that the full text can be imported into literature notes in Obsidian through Zotero Integration, like the Omnivore read-it-later app and Obsidian plugin can. Perhaps Pandoc could be used for this.
